### PR TITLE
Add OneStatementPerLineCheck to Checkstyle rules

### PR DESCRIPTION
**Pull request by cbuescher**
This change adds the OneStatementPerLineCheck to our checkstyle precommit
checks. This rule restricts the number of statements per line to one. The
resoning behind this is that it is very difficult to read multiple statements on
one line. People seem to mostly use more than one statement per line in short 
lambdas and switch statements in our code base, but just going through the 
necessary changes this rule enforces already uncovered some actual problems 
in randomization in test code, so I think its worth it.

### CODE REVIEW

**Comment by cbuescher** [comment]
For context: I opened this as a follow up to https://github.com/elastic/elasticsearch/pull/33676 after I (or better my Eclipse IDE) stumbled into some offending double semicolons in import statements. Those would be complained about as well with this additional checkstyle rule.

File: server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/StringTermsIT.java [hunk 1]
```diff
@@ -87,7 +87,7 @@
 
     @Before
     public void randomizeOptimizations() {
-        TermsAggregatorFactory.COLLECT_SEGMENT_ORDS = false;randomBoolean();
+        TermsAggregatorFactory.COLLECT_SEGMENT_ORDS = randomBoolean();
```
Comment (cbuescher) [inline server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/StringTermsIT.java]: I also opened a separate PR to separate this change: https://github.com/elastic/elasticsearch/pull/33678

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/TokenServiceTests.java [hunk 2]
```diff
@@ -253,7 +253,7 @@ private void rotateKeys(TokenService tokenService) {
 
     public void testKeyExchange() throws Exception {
         TokenService tokenService = new TokenService(tokenServiceEnabledSettings, systemUTC(), client, securityIndex, clusterService);
-        int numRotations = 0;randomIntBetween(1, 5);
+        int numRotations = randomIntBetween(1, 5);
```
Comment (cbuescher) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/TokenServiceTests.java]: I also opened a separate PR for this change: https://github.com/elastic/elasticsearch/pull/33681

**Comment by elasticdog** [comment]
This job triggered CI during a migration of the master. Kicking off an additional build for you manually...

Jenkins, test this please.

File: plugins/mapper-annotated-text/src/main/java/org/elasticsearch/index/mapper/annotatedtext/AnnotatedTextFieldMapper.java [hunk 3]
```diff
@@ -75,14 +75,14 @@
 import static org.elasticsearch.index.mapper.TypeParsers.parseTextField;
 
 /** A {@link FieldMapper} for full-text fields with annotation markup e.g.
- *
```
Comment (nik9000) [inline plugins/mapper-annotated-text/src/main/java/org/elasticsearch/index/mapper/annotatedtext/AnnotatedTextFieldMapper.java]: Could you extract the formatting changes to this file into another commit so I don't have to review them? I'd be happy with you just pushing it so long as all it does is remove trailing spaces.

**Review by nik9000 — COMMENTED** [review]
I kind of like the switch statements the old way but if we can't convince checkstyle that that is ok then I think this is a net win.

File: plugins/mapper-annotated-text/src/main/java/org/elasticsearch/index/mapper/annotatedtext/AnnotatedTextFieldMapper.java (hunk 3 above)
Comment (cbuescher) [reply plugins/mapper-annotated-text/src/main/java/org/elasticsearch/index/mapper/annotatedtext/AnnotatedTextFieldMapper.java]: Sure, I didn't realize this did a reformatting. Will clean this up.

File: server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/StringTermsIT.java (hunk 1 above)
Comment (cbuescher) [reply server/src/test/java/org/elasticsearch/search/aggregations/bucket/terms/StringTermsIT.java]: Actually it wasn't reformatting, just my Auto-Save trailing whitespace removal.

**Comment by cbuescher** [comment]
> I kind of like the switch statements

I can understand that for short lines, but I must say I think readability still suffers. Plus this rule fund the two leftovers in test I mentioned, I think its worth it.

**Review by nik9000 — APPROVED** [review]

**Comment by cbuescher** [comment]
@elasticmachine run gradle test please

**Comment by cbuescher** [comment]
@elasticmachine run gradle check please

**Comment by cbuescher** [comment]
@elasticmachine  run gradle build tests

**Pull request merged**